pkg/controller/network/resourcerecordset/controller.go

```diff
@@ -0,0 +1,179 @@
+		Named(name).
+		For(&v1alpha3.ResourceRecordSet{}).
+		Complete(managed.NewReconciler(mgr,
+			resource.ManagedKind(v1alpha3.ResourceRecordSetGroupVersionKind),
+			managed.WithExternalConnecter(&connector{client: mgr.GetClient(), newClientFn: resourcerecordset.NewClient, awsConfigFn: utils.RetrieveAwsConfigFromProvider}),
+			managed.WithReferenceResolver(managed.NewAPISimpleReferenceResolver(mgr.GetClient())),
+			managed.WithConnectionPublishers(),
+			managed.WithInitializers(),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+type connector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) resourcerecordset.Client
+	awsConfigFn func(context.Context, client.Reader, *corev1.ObjectReference) (*aws.Config, error)
+}
+
+func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha3.ResourceRecordSet)
+	if !ok {
+		return nil, errors.New(errUnexpectedObject)
+	}
+
+	awsconfig, err := conn.awsConfigFn(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c := conn.newClientFn(awsconfig)
+
+	return &external{kube: conn.client, client: c}, nil
+}
+
+type external struct {
+	kube   client.Client
+	client resourcerecordset.Client
+}
+
+func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha3.ResourceRecordSet)
+
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
+	}
+
+	rrset, err := resourcerecordset.GetResourceRecordSet(ctx, e.client, cr.Spec.ForProvider)
+	if _, ok := err.(*resourcerecordset.NotFound); ok {
+		return managed.ExternalObservation{ResourceExists: false}, nil
+	}
+	if err != nil {
+		// Either there is err and retry. Or Resource does not exist.
+		return managed.ExternalObservation{
+			ResourceExists:    false,
+			ConnectionDetails: managed.ConnectionDetails{},
+		}, errors.Wrap(resource.Ignore(zone.IsErrorNoSuchHostedZone, err), errList)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Available())
+
+	upToDate, err := resourcerecordset.IsUpToDate(cr.Spec.ForProvider, rrset)
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(err, errState)
+	}
+
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  upToDate,
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+func (e *external) Create(ctx context.Context, mgd resource.Managed) (managed.ExternalCreation, error) {
+	cr, ok := mgd.(*v1alpha3.ResourceRecordSet)
+	if !ok {
+		return managed.ExternalCreation{}, errors.New(errUnexpectedObject)
+	}
+
+	cr.Status.SetConditions(runtimev1alpha1.Creating())
+
+	input := resourcerecordset.GenerateChangeResourceRecordSetsInput(&cr.Spec.ForProvider, route53.ChangeActionCreate)
+	_, err := e.client.ChangeResourceRecordSetsRequest(input).Send(ctx)
```

Comment:
It's interesting that AWS doesn't really have a unique identifier for this resource. I am wondering whether it should just be an array in `HostedZone` CR. I know Terraform implements it as [distinct resource](https://www.terraform.io/docs/providers/aws/r/route53_record.html) but I can't easily see how we can make it really work. @negz @hasheddan @kasey any thoughts?